src/socket/tcp.rs

```diff
@@ -390,14 +402,23 @@ impl Display for Tuple {
     }
 }
 
+#[cfg(feature = "socket-tcp-cubic")]
+type DefaultCC = congestion_controller::cubic::Cubic;
+
+#[cfg(feature = "socket-tcp-reno")]
+type DefaultCC = congestion_controller::reno::Reno;
+
+#[cfg(not(any(feature = "socket-tcp-cubic", feature = "socket-tcp-reno")))]
+type DefaultCC = congestion_controller::no_control::NoControl;
+
 /// A Transmission Control Protocol socket.
 ///
 /// A TCP socket may passively listen for connections or actively connect to another endpoint.
 /// Note that, for listening sockets, there is no "backlog"; to be able to simultaneously
 /// accept several connections, as many sockets must be allocated, or any new connection
 /// attempts will be reset.
 #[derive(Debug)]
-pub struct Socket<'a> {
+pub struct Socket<'a, CC: CongestionController = DefaultCC> {
```

Comment:
there's no use in having this as a generic param, the user can't set it to something other than default due to how `SocketSet` works. 

I'd instead use the type directly on the field: `congestion_controller: DefaultCC`. And maybe rename it to something like "active" or "chosen": `ActiveCC` or `ActiveCongestionController.`

Reply:
All right.
I will fix to remove the generic parameter, and place it to the field.

Thank you for your review.

Reply:
Prepare `ActiveCC` as follows.
https://github.com/ytakano/smoltcp/blob/179fd7e5c9f552eabd7da3e955d132d2cf4d8b97/src/socket/tcp/congestion_controller.rs#L14-L21

The active congestion controller can be replaced by `socket-tcp-cubic` or `socket-tcp-reno` features.

Reply:
features are removed.
See https://github.com/smoltcp-rs/smoltcp/pull/907#issuecomment-1956049830

Reply:
I think keeping futures makes sense to reduce code size on embedded devices.

Reply:
This is especially important if you're using `f64` since those will cause a large amount of code to be linked in for softfloat.

Reply:
OK.
I will add features again for control congestion control.

Reply:
I have added `socket-tcp-cubic` feature for Cubic.
Cubic is disabled by default.

I think Reno is not bad for small devices like STM32, and uses can use Cubic by specifying the feature.

Fixed-point numbers crate may help the improvement, but is should be another PR I think.
https://docs.rs/fixed/latest/fixed/

Reply:
Inserted a message of Cubic as follows.
```rust
/// To use cubic, please enable the `socket-tcp-cubic` feature in the `smoltcp` crate.
pub fn set_congestion_control(&mut self, congestion_control: CongestionControl) {}
```